example/src/screens/BasicFullscreenHandling.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useCallback, useState } from 'react';
+import { View, Platform, StyleSheet, StatusBar } from 'react-native';
+import { useFocusEffect } from '@react-navigation/native';
+import {
+  Event,
+  usePlayer,
+  PlayerView,
+  SourceType,
+  AudioSession,
+} from 'bitmovin-player-react-native';
+import { useTVGestures } from '../hooks';
+import { FullscreenHandler } from '../../../src/ui/fullscreenhandler';
+
+function prettyPrint(header: string, obj: any) {
+  console.log(header, JSON.stringify(obj, null, 2));
+}
+
+class SampleFullscreenHandler implements FullscreenHandler {
+  isFullscreenActive: boolean = true;
+  onFullscreen: (fullscreenMode: boolean) => void;
+
+  constructor(
+    isFullscreenActive: boolean,
+    onFullscreen: (fullscreenMode: boolean) => void
+  ) {
+    this.isFullscreenActive = isFullscreenActive;
+    this.onFullscreen = onFullscreen;
+  }
+
+  enterFullscreen(): void {
+    this.onFullscreen(true);
+    StatusBar.setHidden(true);
+    this.isFullscreenActive = true;
+    console.log('enter fullscreen');
+  }
+
+  exitFullscreen(): void {
+    this.onFullscreen(false);
+    StatusBar.setHidden(false);
+    this.isFullscreenActive = false;
+    console.log('exit fullscreen');
+  }
+}
+export default function BasicFullscreenHandling({ navigation }) {
+  useTVGestures();
+
+  const player = usePlayer();
+
+  const [fullscreenMode, setFullscreenMode] = useState(false);
+  const fullscreenHandler = new SampleFullscreenHandler(
+    fullscreenMode,
+    (isFullscreen: boolean) => {
+      setFullscreenMode(isFullscreen);
+      navigation.setOptions({ headerShown: !isFullscreen });
+    }
+  );
+  useFocusEffect(
+    useCallback(() => {
+      // iOS audio session must be set to `playback` first otherwise PiP mode won't work.
+      //
+      // Usually it's desireable to set the audio's category only once during your app's main component
+      // initialization. This way you can guarantee that your app's audio category is properly
+      // configured throughout the whole lifecycle of the application.
+      AudioSession.setCategory('playback').catch((error) => {
+        // Handle any native errors that might occur while setting the audio's category.
+        console.log(
+          "[BasicFullscreen] Failed to set app's audio category to `playback`:\n",
+          error
+        );
+      });
+
+      // Load desired source configuration
+      player.load({
+        url:
+          Platform.OS === 'ios'
+            ? 'https://bitmovin-a.akamaihd.net/content/MI201109210084_1/m3u8s/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.m3u8'
+            : 'https://bitmovin-a.akamaihd.net/content/MI201109210084_1/mpds/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.mpd',
```

Comment:
Why don't we use the hls source for Android? just curious.

Reply:
We do this all the time, maybe to showcase we can do DASH on Android.